lib/mapper.js

```diff
@@ -147,15 +147,19 @@ function common(facade){
   if (facade.city()) ret.city = facade.city();
   if (facade.region()) ret.region = facade.region();
 
+//Older segment clients sent idfa, newer ones send advertisingId
   if (os && os.toString().toLowerCase() == 'ios') {
       ret.idfa = facade.proxy('device.idfa');
-  } else if (os && os.toString().toLowerCase() == 'android') {
-      //Older segment clients sent idfa, newer ones send advertisingId
+      if (!ret.idfa) {
+        ret.idfa = facade.proxy('device.advertisingId');
```

Comment:
FWIW this could be simplified to `ret.idfa = facade.proxy('device.idfa') || facade.proxy('device.advertisingId');` (order would define precedence).

It might be better to actually even check `device.advertisingId` before `idfa`.